src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1058,6 +1072,29 @@ static bool isOrContainsErrorType(TypeSymbol typeSymbol)
             }
         }
 
+        /// <summary>
+        /// Return true if <paramref name="overridingReturnType"/> is valid for the return type of an override method when the overridden method's return type is <paramref name="overriddenReturnType"/>.
+        /// </summary>
+        private bool IsValidOverrideReturnType(Symbol overridingSymbol, TypeWithAnnotations overridingReturnType, TypeWithAnnotations overriddenReturnType, DiagnosticBag diagnostics)
+        {
+            if (DeclaringCompilation.LanguageVersion >= MessageID.IDS_FeatureCovariantReturnsForOverrides.RequiredVersion())
```

Comment:
>IDS_FeatureCovariantReturnsForOverrides [](start = 66, length = 39)

Just to confirm, I think we can assume that language version implies newer platform (ie. one that supports covariant returns). If not, we should add a platform check as well

Reply:
We do not support emitting metadata (targeting any plaform) yet. When we do, I'll add a check if apropos.

---
In reply to: [414007929](https://github.com/dotnet/roslyn/pull/43576#discussion_r414007929) [](ancestors = 414007929)